kuba-kostrzewski/simpleToDoApp/src/main/java/pl/kostrzej/simpleToDoApp/app/AppController.java

```diff
@@ -0,0 +1,156 @@
+package pl.kostrzej.simpleToDoApp.app;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Controller;
+import pl.kostrzej.simpleToDoApp.components.login.LoginController;
+import pl.kostrzej.simpleToDoApp.components.task.Task;
+import pl.kostrzej.simpleToDoApp.components.task.TaskController;
+import pl.kostrzej.simpleToDoApp.components.user.User;
+import pl.kostrzej.simpleToDoApp.components.user.UserController;
+
+import java.util.Arrays;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+@Controller
+@Slf4j
+@AllArgsConstructor
+public class AppController {
+
+    private Scanner scanner;
+    private LoginController loginController;
+    private UserController userController;
+    private TaskController taskController;
+
+    public void run() {
+        boolean exit = false;
+        User user = null;
+        while (user == null && !exit){
+            printWelcomeMenu();
+            switch (chooseWelcomeMenuOption()) {
+                case LOGIN:
+                    user = loginController.logIn();
```

Comment:
logIn zwraca usera albo nulla - nie masz rozróżnienia poprawnego logowania od błędnego. W przypadku, gdy logowanie się odbędzie niepoprawnie nie możesz zwracać nulla.

Reply:
Hmm, jak mógłbym to w takim razie rozwiązać? Teraz gdy logowanie się nie odbędzie i użytkownik będzie nullem to program zostaje w pętli związanej z logowaniem/rejestracją i nie przejdzie do pętli z menu głównym dopóki logowanie nie zwróci poprawnego użytkownika.